### PR TITLE
Create MetaDB derived item_statistical_codes

### PR DESCRIPTION
**Pull request by lib-eric**
Create a metadb derived table for statistical codes on items. Table captures the statistical code, the name of the code, and name of the code type.

### CODE REVIEW

**Comment by eliana-flo** [comment]
Looking good! All tests passed for me. 

Query runs
- [X] Data output makes sense
- [X] Data types look correct
- [X] Query logic seems correct
- [X] Column names make sense and are spelled correctly
- [X] Query syntax and style are appropriate (e.g., correct use of double quotes, lower case for data types)

For derived tables:
- [X] query has indexes on columns
- [X] vacuum analyze has been added to the bottom
- [X] table appears in runlist.txt at an appropriate point and with correct file name

**Review by eliana-flo — APPROVED** [review]
Looks like I did things a little out of order. :)

File: sql_metadb/derived_tables/item_statistical_codes.sql [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        statistical_code_ids.data #>> '{}' AS statistical_code_id
+    FROM
+        folio_inventory.item AS i
+        CROSS JOIN jsonb_array_elements(jsonb_extract_path(jsonb, 'statisticalCodeIds')) AS statistical_code_ids(data)
+        LEFT JOIN folio_inventory.item__t AS it ON i.id = it.id
+)
+SELECT
+    items_statistical_codes.item_id,
+    items_statistical_codes.item_hrid,
+    items_statistical_codes.statistical_code_id,
```
Comment (stdombek) [inline sql_metadb/derived_tables/item_statistical_codes.sql:18]: Should be of type uuid.

File: sql_metadb/derived_tables/item_statistical_codes.sql [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    sc__t.code AS statistical_code,
+    sc__t.name AS statistical_code_name,
+    sc__t.statistical_code_type_id,
```
Comment (stdombek) [inline sql_metadb/derived_tables/item_statistical_codes.sql:21]: Should be of type uuid.

**Review by stdombek — CHANGES_REQUESTED** [review]
- [X] Query runs
- [X] Data output makes sense
- [ ] Data types look correct
- [X] Query logic seems correct
- [X] Column names make sense and are spelled correctly
- [X] Query syntax and style are appropriate (e.g., correct use of double quotes, lower case for data types)

For derived tables:
- [X] query has indexes on columns
- [X] vacuum analyze has been added to the bottom
- [X] table appears in runlist.txt at an appropriate point

File: sql_metadb/derived_tables/item_statistical_codes.sql (hunk 1 above)
Comment (lib-eric) [reply sql_metadb/derived_tables/item_statistical_codes.sql]: Cast `statistical_code_id` to `uuid`

**Review by nassibnassar — CHANGES_REQUESTED** [review]
Could we get a clearer summary in the PR description?

**Comment by adoerrer** [comment]
> Could we get a clearer summary in the PR description?

I'm not sure if this has happened already. @lib-eric could you please take a look on that? Thanks!

**Pull request merged**